src/kernels/internalTypes.ts

```diff
@@ -11,7 +11,7 @@ export enum ContributedKernelFinderKind {
 }
 
 export interface IContributedKernelFinder<T extends KernelConnectionMetadata = KernelConnectionMetadata> {
-    status: 'discovering' | 'idle';
+    readonly status: 'discovering' | 'idle';
     onDidChangeStatus: Event<void>;

```

Comment:
readonly too?

Reply:
Yes, but these are old interfaces/code that is slowly being replaced by another interface, in that all are readonly and this stats is the only overlapping property, and I had to make this reaodnly to ensure those two interfaces are compatible.

Basically not too keen on adding `readonly` to all of the others as its pointless as they will be phased out and removed.
Hope thats ok.